python/ray/train/xgboost/xgboost_trainer.py

```diff
@@ -138,16 +188,21 @@ class XGBoostTrainer(SimpleXGBoostTrainer):
     def __init__(
         self,
         *,
-        datasets: Dict[str, GenDataset],
-        label_column: str,
-        params: Dict[str, Any],
-        dmatrix_params: Optional[Dict[str, Dict[str, Any]]] = _DEPRECATED_VALUE,
-        num_boost_round: int = 10,
+        train_loop_per_worker: Optional[
+            Union[Callable[[], None], Callable[[Dict], None]]
+        ] = None,
+        train_loop_config: Optional[Dict] = None,
+        xgboost_config: Optional[XGBoostConfig] = None,
         scaling_config: Optional[ray.train.ScalingConfig] = None,
         run_config: Optional[ray.train.RunConfig] = None,
+        datasets: Optional[Dict[str, GenDataset]] = None,
         dataset_config: Optional[ray.train.DataConfig] = None,
         resume_from_checkpoint: Optional[Checkpoint] = None,
         metadata: Optional[Dict[str, Any]] = None,
+        # TODO(justinvyu): [Deprecated] Legacy XGBoostTrainer API
+        label_column: Optional[str] = None,
+        params: Optional[Dict[str, Any]] = None,
+        num_boost_round: Optional[int] = None,
```

Comment:
Note: it's fine to change the ordering of these params because they were already forced to be kwargs.